### PR TITLE
Introduce hooks api to react-redux

### LINKED ISSUE #441: Move react-redux usage of connect for stateful screens/components over to hooks API

### PR DESCRIPTION
**Pull request by pedrobergamini**
### Issue link

`https://github.com/tasitlabs/tasit/issues/441`

### Auto-close the issue?

Closes #441 

### Types of changes

Technical debt (a code change that doesn't fix a bug or add a feature but makes something clearer for devs)

### Notes
Removes usage of react-redux connect for introducing the hooks API.

**I created a GlobalState interface that requires further work for using reducers types**

### CODE REVIEW

**Comment by pcowgill** [comment]
@pedrobergamini Thanks so much for working on this!

When you said, "I created a GlobalState interface that requires further work for using reducers types", is that
(A) something that would cause problems when running the app now
or
(B) just something we'd have to keep in mind if we wanted to make further changes
?

**Comment by pcowgill** [comment]
Oh, was it this TODO you had in mind when saying that? https://github.com/pedrobergamini/tasit/blob/46b3b17a62a96264f022dc6b316cf00abc908b57/decentraland/types/GlobalState.ts#L1

In that case, it's in category (B) above.

**Comment by pedrobergamini** [comment]
Exactly, it's in category B

File: decentraland/screens/BuyLandScreen/BuyLandScreen.tsx [hunk 1]
```diff
@@ -24,32 +26,31 @@ interface AccountInfoObject {
   creationActions: any[];
 }
 
+interface SelectedState {
+  accountInfo: AccountInfoObject;
+  landToBuy: any;
```
Comment (pcowgill) [inline decentraland/screens/BuyLandScreen/BuyLandScreen.tsx:31]:
```suggestion
  landToBuy: any;
  myAssets: any[];
```

File: decentraland/screens/BuyLandScreen/BuyLandScreen.tsx [hunk 2]
```diff
@@ -24,32 +26,31 @@ interface AccountInfoObject {
...
 }) => {
+  const dispatch = useDispatch();
+
+  const { accountInfo, landToBuy: landForSale } = useSelector<
+    GlobalState,
+    SelectedState
+  >(state => {
+    const { accountInfo, landToBuy, myAssets } = state;
+    const { list: myAssetsList } = myAssets;
+    return { accountInfo, landToBuy, myAssets: myAssetsList };
```
[the remark this answers is not in the document]
Comment (pcowgill) [reply decentraland/screens/BuyLandScreen/BuyLandScreen.tsx:51]: I think the necessary change may be as simple as accepting this suggestion https://github.com/tasitlabs/tasit/pull/476#discussion_r412311484

**Pull request merged**